sdk/playwrighttesting/microsoft-playwright-testing/src/core/playwrightService.ts

```diff
@@ -0,0 +1,158 @@
+
+import { ServiceAuthenticationConstants } from "../common/constants";
+import customerConfig from "../common/customerConfig";
+import { PlaywrightServiceConfig } from "../common/playwrightServiceConfig";
+import playwrightServiceEntra from "./playwrightServiceEntra";
+import {
+  PlaywrightServiceAdditionalOptions,
+  PlaywrightConfig,
+  PlaywrightConfigInput,
+  BrowserConnectOptions,
+} from "../common/types";
+import {
+  emitReportingUrl,
+  fetchOrValidateAccessToken,
+  getAccessToken,
+  getServiceWSEndpoint,
+  validateMptPAT,
+  validateServiceUrl,
+} from "../utils/utils";
+
+/**
+ * @public
+ *
+ * Generate playwright configuration integrated with Microsoft Playwright Testing.
+ *
+ * @param config - base playwright configuration
+ * @param options - additional options for the service
+ * @returns PlaywrightConfig
+ *
+ * @example
+ * ```
+ * import { defineConfig } from "playwright/test";
+ * import { getServiceConfig } from "@azure/microsoft-playwright-testing";
+ * import playwrightConfig from "./playwright.config";
+ *
+ * export default defineConfig(playwrightConfig, getServiceConfig(playwrightConfig));
+ * ```
+ *
+ * @example
+ * ```
+ * import { defineConfig } from "playwright/test";
+ * import { getServiceConfig, ServiceOsConstants } from "@azure/microsoft-playwright-testing";
+ * import playwrightConfig from "./playwright.config";
+ *
+ * export default defineConfig(playwrightConfig, getServiceConfig(playwrightConfig, {
+ *  runId: "custom run id",
+ *  os: ServiceOsConstants.WINDOWS
+ * }));
+ * ```
+ */
+const getServiceConfig = (
+  config: PlaywrightConfigInput,
+  options?: PlaywrightServiceAdditionalOptions,
+): PlaywrightConfig => {
+  validateServiceUrl();
+  if (options?.credential) {
+    playwrightServiceEntra.entraIdAccessToken = options.credential;
+  }
+  if (config.globalSetup) {
+    customerConfig.globalSetup = config.globalSetup;
+  }
+  if (config.globalTeardown) {
+    customerConfig.globalTeardown = config.globalTeardown;
+  }
+
+  const playwrightServiceConfig = new PlaywrightServiceConfig();
+  playwrightServiceConfig.setOptions(options);
+  emitReportingUrl();
+
+  const globalFunctions: any = {};
+  if (options?.defaultAuth === ServiceAuthenticationConstants.SERVICE_TOKEN && getAccessToken()) {
+    // mpt PAT requested and set by the customer, no need to setup entra lifecycle handlers
+    validateMptPAT();
+  } else {
+    globalFunctions.globalSetup = require.resolve("./global/playwright-service-global-setup");
+    globalFunctions.globalTeardown = require.resolve("./global/playwright-service-global-teardown");
+  }
+
+  if (options?.useCloudHostedBrowsers === false) {
+    return {
+      ...globalFunctions,
+    };
+  }
+
+  return {
+    use: {
+      connectOptions: {
+        wsEndpoint: getServiceWSEndpoint(
+          playwrightServiceConfig.runId,
+          playwrightServiceConfig.serviceOs,
+        ),
+        headers: {
+          Authorization: `Bearer ${getAccessToken()}`,
```

Comment:
NIT: something to think, if we should allow customer to pass more headers ? for use case like channel browser, etc as this code is now doing into package.

Reply:
https://playwright.dev/docs/api/class-browsertype#browser-type-connect - The docs don't mention any parameters, plus it will be handled by OSS.

Reply:
What about these x-playwright-launch-options ?